Yank/yank.py

```diff
@@ -245,6 +251,237 @@ def get_region(self, region_name):
+
+        * Single integer (a bit redundant)
+        * Iterable of integer (also a bit redundant)
+        * Complex Region selection
+        * MDTraj String
+        * **Future Feature** SMARTS Selection
+
+        This method will never return duplicate atom numbers.
+
+        The ``sort_by`` method controls how the output is sorted before being returned, see the details in the
+        Parameters block for information about each option
+
+        The Complex Region string returns an the set of atoms derived from the arguments using logical operators
+        ``and`` and ``or``
+        along with grouping through parenthesis . For example, assume you
+        have two regions ``regionA = [0,1,2,3]`` and ``regionB = [2,3,4,5]``. You can do operations such as the
+        following:
+
+             ``regionA and regionB`` yields ``[2,3]``, which is the intersection of the regions.
+
+             ``regionA or regionB`` yields ``[0,1,2,3,4,5]``, which is the union of the regions.
+
+        More complex statements with more regions will also work, and statements can be grouped with ``()``.
+
+        Parameters
+        ----------
+        selection : str or
+            String defining the selection
+        as_set : bool, Default False
+            Determines output format. Returns a Set if True, otherwise output is a list.
+        sort_by : str or None, Default: 'auto'
+            Determine how to sort the output if ``as_set`` is False.
+
+            * 'auto': Let the selection string determine how to sort it out based on its priorities
+            * 'ordered': Atoms are sorted index, smallest to largest
+            * 'inferred_by_region': Atoms are sorted by which region in the provided ``selection_string`` occurs first.
+                So if your expression is ``region1 and region2``, then the output will be atoms which appear in
+                ``region1`` first. Parenthesis are **ignored** so the expression ``region1 and (region2 or region3)``
+                will prioritize ``region1`` first for sorting. This option only works for expressions on Regions,
+                other selections will fall back to ``None``
+            * ``None``: Sorting is left up to however the selection string is processed by their respective drivers,
+                or by whatever order the set operations returns it. Not guaranteed to be deterministic in all cases.
+
+        Returns
+        -------
+        selection : list or set
+            Returns the selected atoms as either a list or set based on ``as_set`` keyword.
+            Order of the output is determined by the ``sort_by`` keyword. If a ``as_set`` is ``True``, this option has
+            no effect.
+
+        """
+
+        # Helper functions for handling the sorting
+        def sort_output_ordered(sortable):
+            # Dont do list.sort, its an in place action.
+            return sorted(list(sortable))
+
+        def sort_output_inferred_by_region(sortable):
```

Comment:
I had some troubles following the logic here. Why not something like
```python
def sort_output_inferred_by_region(sortable):
    # Only valid when selection is a string
    final_output = []
    region_order = [region_name for region_name in selection.split() if region_name in self]
    for region_name in region_order:
        region = self.get_region(region_name)
        for atom_index in region:
            if atom_index not in final_output:
                final_output.append(atom_index)
```

Reply:
Very helpful, my brain was stuck on the more complex solution because of something I had to do elsewhere